src/components/textarea/textarea.tsx

```diff
@@ -0,0 +1,498 @@
+        <textarea
+          aria-invalid={toAriaBoolean(this.value?.length > this.maxLength)}
+          aria-label={getLabelText(this)}
+          autofocus={this.autofocus}
+          class={{
+            textarea: true,
+            [CSS.resizeDisabled]: this.resizeDisabled,
+            [CSS.resizeDisabledX]: this.horizantalResizeDisabled,
+            [CSS.resizeDisabledY]: this.verticalResizeDisabled,
+            [CSS.readonly]: this.readonly,
+            [CSS.textareaInvalid]: this.value?.length > this.maxLength,
+            [CSS.footerSlotted]: !!this.endSlotElements?.length && !!this.startSlotElements?.length,
+            [CSS.borderColor]: !hasFooter,
+            [CSS.blocksizeFull]: !hasFooter
+          }}
+          cols={this.cols}
+          disabled={this.disabled}
+          name={this.name}
+          onChange={this.handleChange}
+          onInput={this.handleInput}
+          placeholder={this.placeholder}
+          readonly={this.readonly}
+          ref={this.setTextareaEl}
+          required={this.required}
+          rows={this.rows}
+          value={this.value}
+          wrap={this.wrap}
+        />
+        <span class="content">
+          <slot onSlotchange={this.contentSlotChangeHandler} />
+        </span>
+        {
+          <footer
+            class={{
+              [CSS.footer]: true,
+              [CSS.readonly]: this.readonly,
+              [CSS.hide]: !hasFooter
+            }}
+            key={CSS.footer}
+            ref={(el) => (this.footerEl = el as HTMLElement)}
+          >
+            <div class="slot-container">
+              <slot name={SLOTS.footerStart} onSlotchange={this.footerStartSlotChangeHandler} />
+              <slot name={SLOTS.footerEnd} onSlotchange={this.footerEndSlotChangeHandler} />
+            </div>
+            {this.renderCharacterLimit()}
+          </footer>
+        }
+        <HiddenFormInputSlot component={this} />
+      </Host>
+    );
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  //--------------------------------------------------------------------------
+
+  /** Sets focus on the component. */
+  @Method()
+  async setFocus(): Promise<void> {
+    await componentLoaded(this);
+    this.textareaEl.focus();
+  }
+
+  /** Selects all text of the component's `value`. */
+  @Method()
+  async selectText(): Promise<void> {
+    this.textareaEl.select();
+  }
+  //--------------------------------------------------------------------------
+  //
+  //  Private Properties/ State
+  //
+  //--------------------------------------------------------------------------
+
+  formEl: HTMLFormElement;
+
+  defaultValue: Textarea["value"];
+
+  labelEl: HTMLCalciteLabelElement;
+
+  textareaEl: HTMLTextAreaElement;
+
+  footerEl: HTMLElement;
+
+  @State() effectiveLocale = "";
+
+  @State() endSlotElements: Element[];
+
+  @State() startSlotElements: Element[];
+
+  @Watch("effectiveLocale")
```

Comment:
Should this be grouped with the other Watchers?

Reply:
That makes sense! Based on your comment I now see it is watching a State variable. I honestly haven't seen Watchers used for State before, only Props, so I presumed this was for a Prop when I made my suggestion. :)